src/object.c

```diff
@@ -1172,32 +1172,27 @@ struct redisMemOverhead *getMemoryOverheadData(void) {
 
     mem_total += server.initial_memory_usage;
 
-    mem = 0;
-    if (server.repl_backlog && server.repl_backlog->ref_repl_buf_node) {
-        replBufBlock *head = listNodeValue(server.repl_backlog->ref_repl_buf_node);
-        replBufBlock *last = listNodeValue(listLast(server.repl_buffer_blocks));
-        /* The memory of referenced replication buffer blocks. */
-        mem += server.repl_backlog->histlen + (last->size - last->used);
-        /* The linked list node memory and replBufBlock struct memory. */
-        mem += (last->id - head->id + 1) * (sizeof(replBufBlock)+sizeof(listNode));
-        /* The approximate memory of rax tree for indexed blocks. */
-        mem += server.repl_backlog->blocks_index->numnodes * sizeof(raxNode) +
-               raxSize(server.repl_backlog->blocks_index) * sizeof(void*);
```

Comment:
maybe we still want to use the old way of computing the total memory (rax, nodes and all), and then apply the if-else logic on the result?

Reply:
in current code, we consider block list into `mem_clients_slaves`, so we add the rax  into backlog memory.